### PR TITLE
Disable interpreting of strings as DateTime during deserialization

### PR DESCRIPTION
**Pull request by tmat**
Fixes https://github.com/dotnet/roslyn/issues/39429.

### CODE REVIEW

**Comment by tmat** [comment]
@vatsalyaagrawal For 16.7 approval

File: src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs [hunk 1]
```diff
@@ -52,8 +51,11 @@ public async Task UpdaterService()
             // make sure client is ready
             using var client = await service.TryGetRemoteHostClientAsync(CancellationToken.None);
 
-            // add solution
+            // add solution, change document
             workspace.AddSolution(SolutionInfo.Create(SolutionId.CreateNewId(), VersionStamp.Default));
+            var project = workspace.AddProject("proj", LanguageNames.CSharp);
+            var document = workspace.AddDocument(project.Id, "doc.cs", SourceText.From("code"));
+            workspace.ApplyTextChanges(document.Id, new[] { new TextChange(new TextSpan(0, 1), "abc") }, CancellationToken.None);
```
Comment (dibarbet) [inline src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs:58]: Should we add a date-like string text change to the test as well to verify that we don't hit the exception?

**Comment by jasonmalinowski** [comment]
Wait, so if the change looks like a date, then we'd be sending that as the string of the diff and that'd break things? Is there any other settings we should be changing here too?

File: src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs (hunk 1 above)
Comment (tmat) [reply src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs:58]: Tired but doesn't repo in simple cases :( Don't know what exactly triggers the conversion.

**Comment by tmat** [comment]
@jasonmalinowski Correct. @AArnott is not aware of any other switches.

File: src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs:58]: The original bug had `2020-01-15T00:00:00.000Z` in it -- did that not work?

Comment (tmat) [reply src/VisualStudio/Core/Test.Next/Remote/RemoteHostClientServiceFactoryTests.cs:58]: It did not.

**Pull request merged**